### PR TITLE
tools: migrate data from AWS Aurora using DM

### PR DESCRIPTION
**Pull request by csuzhangxc**
@GregoryIan @lilin90 PTAL

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@lilin90 `tools/dm/practice.md` seems not linked from any other docs?

**Comment by lilin90** [comment]
@csuzhangxc That file is in the TOC now.

File: tools/dm/from-aurora.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+| 集群 | 终端节点 | 端口 | 角色 |
+|:-------- |:--- | :--- | :--- |
+| Aurora-1 | pingcap-1.h8emfqdptyc4.us-east-2.rds.amazonaws.com | 3306 | 写入器 |
+| Aurora-1 | pingcap-1-us-east-2a.h8emfqdptyc4.us-east-2.rds.amazonaws.com | 3306 | 读取器 |
+| Aurora-2 | pingcap-2.h8emfqdptyc4.us-east-2.rds.amazonaws.com | 3306 | 写入器 |
+
+DM 在增量同步阶段依赖 `ROW` 格式的 binlog，如果未启用 binlog 及设置正确的 binlog 格式，则不能正常使用 DM 进行数据同步，具体可参见[检查内容](/tools/dm/precheck.md#检查内容)。
+
+> 注意：Aurora 读取器不能开启 binlog，因此不能作为 DM 数据迁移时的上游 master server。
```
Comment (lilin90) [inline tools/dm/from-aurora.md]: The following format is to trigger the background color for Note.

```suggestion
> **注意：**
>
> Aurora 读取器不能开启 binlog，因此不能作为 DM 数据迁移时的上游 master server。
```

File: tools/dm/from-aurora.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+如果需要基于 GTID 进行数据迁移，还需要为 Aurora 集群启用 GTID 支持。
+
+> 注意：基于 GTID 的数据迁移需要 MySQL 5.7 (Aurora 2.04.1) 或更高版本。
```
Comment (lilin90) [inline tools/dm/from-aurora.md]:
```suggestion
> **注意：**
>
> 基于 GTID 的数据迁移需要 MySQL 5.7 (Aurora 2.04.1) 或更高版本。
```

File: tools/dm/from-aurora.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+### 为 Aurora 集群修改 binlog 相关参数
+
+在 Aurora 集群中，binlog 相关参数是**集群参数组中的集群级参数**，有关如何为 Aurora 集群启用 binlog 支持，请参考[在复制主实例上启用二进制日志记录](https://docs.aws.amazon.com/zh_cn/AmazonRDS/latest/AuroraUserGuide/AuroraMySQL.Replication.MySQL.html#AuroraMySQL.Replication.MySQL.EnableBinlog)。在使用 DM 进行数据迁移时，需要将 `binlog_format` 参数设置为 `ROW`。
+
+如果需要基于 GTID 进行数据迁移，需要将 `gtid-mode` 及 `enforce_gtid_consistency` 均设置为 `ON`。 有关如何为 Aurora 集群启用基于 GTID 的数据迁移支持，请参考 [Configuring GTID-Based Replication for an Aurora MySQL Cluster](https://docs.aws.amazon.com/zh_cn/AmazonRDS/latest/AuroraUserGuide/mysql-replication-gtid.html#mysql-replication-gtid.configuring-aurora)。
```
Comment (lilin90) [inline tools/dm/from-aurora.md]:
```suggestion
如果需要基于 GTID 进行数据迁移，需要将 `gtid-mode` 与 `enforce_gtid_consistency` 均设置为 `ON`。有关如何为 Aurora 集群启用基于 GTID 的数据迁移支持，请参考 [Configuring GTID-Based Replication for an Aurora MySQL Cluster](https://docs.aws.amazon.com/zh_cn/AmazonRDS/latest/AuroraUserGuide/mysql-replication-gtid.html#mysql-replication-gtid.configuring-aurora)。
```

File: tools/dm/from-aurora.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+> 注意：在 Aurora 管理后台中，`gtid_mode` 参数表示为 `gtid-mode`。
```
Comment (lilin90) [inline tools/dm/from-aurora.md]:
```suggestion
> **注意：**
>
> 在 Aurora 管理后台中，`gtid_mode` 参数表示为 `gtid-mode`。
```

File: tools/dm/from-aurora.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+3. 执行以下命令启动数据同步任务
+
+    ```bash
+    # `task.yaml` 是之前编辑的配置文件
+    start-task ./task.yaml
+    ```
+    
+    - 如果执行命令后的返回结果中不包含错误信息，则表明任务已经成功启动
+    - 如果包含以下错误信息，则表明上游 Aurora 用户可能拥有 TiDB 不支持的权限类型
+        ```json
```
Comment (lilin90) [inline tools/dm/from-aurora.md]: Add a blank line before a code block.

```suggestion
        
        ```json
```

File: tools/dm/from-aurora.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+            "id": 5,
+            "name": "source db replication privilege chcker",
+            "desc": "check replication privileges of source DB",
+            "state": "fail",
+            "errorMsg": "line 1 column 285 near \"LOAD FROM S3, SELECT INTO S3 ON *.* TO 'root'@'%' WITH GRANT OPTION\" ...",
+            "instruction": "",
+            "extra": "address of db instance - pingcap-1.h8emfqdptyc4.us-east-2.rds.amazonaws.com"
+        }
+        ```
+        此时可以选择以下两种处理方法中的任意一种进行处理后，再使用 `start-task` 尝试重新启动任务：
```
Comment (lilin90) [inline tools/dm/from-aurora.md]:
```suggestion
        
        此时可以选择以下两种处理方法中的任意一种进行处理后，再使用 `start-task` 尝试重新启动任务：
```

File: tools/dm/from-aurora.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+        1. 为用于进行数据迁移的 Aurora 用户移除不被 TiDB 支持的不必要的权限
+        2. 如果能确保 Aurora 用户拥有 DM 所需要的权限，可以在 task.yaml 配置文件中添加如下顶级配置项以跳过启用任务时的前置权限检查
```
Comment (lilin90) [inline tools/dm/from-aurora.md]:
```suggestion
        2. 如果能确保 Aurora 用户拥有 DM 所需要的权限，可以在 `task.yaml` 配置文件中添加如下顶级配置项以跳过启用任务时的前置权限检查
```

File: tools/dm/from-aurora.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+## 第 6 步：查询任务
+
+如需了解 DM 集群中是否存在正在运行的同步任务及任务状态等信息，可在 dmctl 内使用以下命令进行查询：
+
+```bash
+query-status
+```
+
+> 注意：
```
Comment (lilin90) [inline tools/dm/from-aurora.md]:
```suggestion
> **注意：**
>
```

File: tools/dm/from-aurora.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+> 如果查询命令的返回结果中包含以下错误信息，则表明在全量同步的 dump 阶段不能获得相应的 lock：
+>   ```bash
```
Comment (lilin90) [inline tools/dm/from-aurora.md]: I'm afraid that the code block cannot be displayed correctly on the PingCAP website, although it looks well on GitHub. We can keep it and have a try. If it doesn't work, could we change it into inline code?

File: tools/dm/from-aurora.md [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+    | 数据库实例 | 主机 | 端口 | 用户名 | 加密密码 |
+    |:-------- |:--- | :--- | :--- | :--- |
+    | 上游 Aurora-1 | pingcap-1.h8emfqdptyc4.us-east-2.rds.amazonaws.com | 3306 | root | VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU= |
+    | 上游 Aurora-2 | pingcap-2.h8emfqdptyc4.us-east-2.rds.amazonaws.com | 3306 | root | VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU= |
+    | 下游 TiDB | 172.16.10.83 | 4000 | root | |
+
+- dm-master 进程配置文件 `{ansible deploy}/conf/dm-master.toml` 中的配置
+
+    ```toml
+    # Master 配置
```
Comment (yikeke) [inline tools/dm/from-aurora.md]:
```suggestion
    # Master 配置。
```

File: tools/dm/from-aurora.md (hunk 9 above)
Comment (csuzhangxc) [reply tools/dm/from-aurora.md]: If it doesn't worker, we could change it into inline code.

**Comment by csuzhangxc** [comment]
@lilin90 @yikeke Thanks a lot, PTAL again.

File: tools/dm/from-aurora.md (hunk 9 above)
Comment (lilin90) [reply tools/dm/from-aurora.md]: @csuzhangxc OK.

**Comment by lilin90** [comment]
@GregoryIan PTAL

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**